app/src/main/java/com/example/diceroller/DiceRollerViewModel.kt

```diff
@@ -0,0 +1,63 @@
+package com.example.diceroller
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.getAndUpdate
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.launch
+import java.util.Calendar
+import kotlin.random.Random
+
+private const val REPLAY_LAST: Int = 1
+
+class DiceRollerViewModel : ViewModel() {
```

Comment:
Because of a rename this is looking like a new file. This is actually your original ViewModel with some light work.